pkg/istio-agent/agent.go

```diff
@@ -494,23 +494,24 @@ func (a *Agent) Run(ctx context.Context) (func(), error) {
 
 func (a *Agent) initSdsServer() error {
 	var err error
-	if security.CheckWorkloadCertificate(security.WorkloadIdentityCertChainPath, security.WorkloadIdentityKeyPath, security.WorkloadIdentityRootCertPath) {
+	// If proxy is using file mounted certs, we do not have to connect to CA.
+	if a.secOpts.FileMountedCerts {
```

Comment:
This "if ... else if" changes the program behavior, not just fixes logs. This "if ... else if" leads to the following problem: when workload certificate is used, FileMountedCerts can be true, which leads to the following three variables not being set.
                 a.secOpts.RootCertFilePath = security.WorkloadIdentityRootCertPath
		a.secOpts.CertChainFilePath = security.WorkloadIdentityCertChainPath
		a.secOpts.KeyFilePath = security.WorkloadIdentityKeyPath

The correct behavior is to first CheckWorkloadCertificate(security.WorkloadIdentityCertChainPath, security.WorkloadIdentityKeyPath, security.WorkloadIdentityRootCertPath), then check a.secOpts.FileMountedCerts:
if security.CheckWorkloadCertificate(security.WorkloadIdentityCertChainPath, security.WorkloadIdentityKeyPath, security.WorkloadIdentityRootCertPath) {
		log.Info("workload certificate files detected, creating secret manager without caClient")
		a.secOpts.RootCertFilePath = security.WorkloadIdentityRootCertPath
		a.secOpts.CertChainFilePath = security.WorkloadIdentityCertChainPath
		a.secOpts.KeyFilePath = **security.WorkloadIdentityKeyPath**
                 ....
} else if a.secOpts.FileMountedCerts {
....
}

Reply:
I started with log changes only but thought it would be better to handle both fileMounted and checkWorkloadCertificate in the same manner at the top to avoid file checking unnecessarily for file mounted certs flow- Sorry I should have updated the PR.

I think it does not change behaviour because both are not typically used at the same time. But I do not have strong opinion on that. So reverted the change

Reply:
Thanks! The PR looks much better now.